src/main/java/software/amazon/event/kafkaconnector/EventBridgeResult.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+import software.amazon.event.kafkaconnector.util.MappedSinkRecord;
+
+public class EventBridgeResult<T> {
+
+  public static class Error {
+
+    private final String message;
+    private final int level;
+
+    private Error(String message, int level) {
+      this.message = message;
+      this.level = level;
+    }
+
+    public static Error of(String message, int level) {
+      return new Error(message, level);
+    }
+
+    public String getMessage() {
+      return message;
+    }
+
+    public int getLevel() {
+      return level;
+    }
+  }
+
+  private static class Failure {
+
+    private final MappedSinkRecord<Error> value;
+
+    Failure(SinkRecord sinkRecord, Error error) {
+      value = new MappedSinkRecord<>(sinkRecord, error);
+    }
+  }
+
+  private final Object value;
+
+  private EventBridgeResult(Object value) {
+    this.value = value;
+  }
+
+  public static <T> EventBridgeResult<T> success(SinkRecord sinkRecord, T value) {
+    return new EventBridgeResult<>(new MappedSinkRecord<>(sinkRecord, value));
+  }
+
+  public static <T> EventBridgeResult<T> failure(SinkRecord sinkRecord, Error error) {
+    return new EventBridgeResult<>(new Failure(sinkRecord, error));
+  }
+
+  public static <T> EventBridgeResult<T> failure(MappedSinkRecord<Error> value) {
+    return new EventBridgeResult<>(new Failure(value.sinkRecord, value.value));
+  }
+
+  public boolean isSuccess() {
+    return value == null || !Failure.class.equals(value.getClass());
```

Comment:
Question: IMHO even for success we get EventBridgeResult so why check `value == null` here? Perhaps needs some comments

Reply:
We have to check for `value != null` to get it's class `value.getClass()`. It's the simplification of `(value != null && !Failure.class.equals(value.getClass())) || (value == null)`.

Reply:
So that means any null value will could be considered success from this code path? Again, we should doc this behavior since it's public (at least exposed in our code base)